rust/src/shards/gui/widgets/image.rs

```diff
@@ -156,23 +156,10 @@ impl Shard for Image {
 impl Image {
   fn activateImage(&mut self, _context: &Context, input: &Var) -> Result<Var, &str> {
     if let Some(ui) = util::get_current_parent(*self.parents.get())? {
-      let shimage: &SHImage = input.try_into()?;
-      let ptr = shimage.data;
-      let texture = if ptr != self.prev_ptr {
-        let image: egui::ColorImage = shimage.into();
-        self.prev_ptr = ptr;
-        self.texture.insert(ui.ctx().load_texture(
-          format!("UI.Image: {:p}", shimage.data),
-          image,
-          Default::default(),
-        ))
-      } else {
-        self.texture.as_ref().unwrap()
-      };
-
-      let scale: (f32, f32) = self.scale.get().try_into()?;
-      let scale: egui::Vec2 = scale.into();
-      ui.image(texture, scale * texture.size_vec2());
+      let texture = image_util::ui_image_cached(&mut self.cached_ui_image, input, ui)?;
```

Comment:
what's cached?

Reply:
The loaded egui texture if you create it from a shards image, same as previous behavior but I wrapped it so it could be reused for image_button, etc.